service/storage_proxy.hh

```diff
@@ -796,7 +799,7 @@ public:
     future<bool> cas(schema_ptr schema, cas_shard cas_shard, shared_ptr<cas_request> request, lw_shared_ptr<query::read_command> cmd,
             dht::partition_range_vector partition_ranges, coordinator_query_options query_options,
             db::consistency_level cl_for_paxos, db::consistency_level cl_for_learn,
-            clock_type::time_point write_timeout, clock_type::time_point cas_timeout, bool write = true);
+            clock_type::time_point write_timeout, clock_type::time_point cas_timeout, bool write = true, std::optional<cdc::per_request_options> cdc_opts = {});
```

Comment:
Looking at this PR, and superficially at #26121, do we gain much by using `std::optional`? If I'm not mistaken, the default options for that struct should be equal to the case where the optional is disengaged, no? Using `std::optional` does not cause us to save anything with regards to stack space, on the contrary - you need to maintain a flag which indicates whether the optional is valid or not.